04-store-api/starter/app.js

```diff
@@ -1 +1,38 @@
-console.log('04 Store API')
+require("dotenv").config();
+require("express-async-errors");
+
+const express = require("express");
+const app = express();
+
+const connectDB = require("./db/connect.js");
+
+const productsRouter = require("./routes/products.js");
+
+const notFoundMiddleware = require("./middleware/not-found.js");
+const errorMiddleware = require("./middleware/error-handler.js");
+
+app.use(express.json());
+
+//routes
+
+app.get("/", (req, res) => {
+	res.send('<h1>Store API</h1><a href="/api/v1/products">products route</a>');
+});
+
+app.use("/api/v1/products", productsRouter);
+// product route
+
+app.use(notFoundMiddleware);
+app.use(errorMiddleware);
+
+const port = process.env.PORT || 3000;
+
+const start = async () => {
+	try {
+		await connectDB(process.env.MONGO_URI);
```

Comment:
In order to use `await` on `connectDB()` it needs to be able to return a promise.

Reply:
i think mongoose.connect is a promise https://mongoosejs.com/docs/connections.html